runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/SdkHarnessClient.java

```diff
@@ -384,7 +382,7 @@ public SdkHarnessClient withIdGenerator(IdGenerator idGenerator) {
         BeamFnApi.ProcessBundleDescriptor.class.getSimpleName(),
         descriptor.getId(),
         BeamFnApi.ProcessBundleDescriptor.class.getSimpleName());
-    return bundleProcessor;
+    return (BundleProcessor<T>) bundleProcessor;
```

Comment:
This is actually an unchecked cast too, but I think we have the warning disabled globally. You can suppress the warning on line 372 if you want to move towards -Wall -Werror (my favorite setting :-)

Reply:
Makes sense. Moving the unchecked casting at line 372 and suppressing it.